src/main/java/io/candydoc/ddd/extract_domain_vocabulary/ReflectionsDomainVocabularyFinder.java

```diff
@@ -161,10 +163,34 @@ private static Set<FullName> dependenciesOf(Class<?> domainModel, Set<Class<?>>
     return dependenciesOfTacticalPattern(domainModel);
   }
 
+  private static List<Class<?>> getFieldType(Field field) {
+    return Optional.ofNullable(field.getGenericType())
+        .filter(ParameterizedType.class::isInstance)
+        .map(ReflectionsDomainVocabularyFinder::extractTypeParameters)
+        .orElse(List.of(field.getType()));
+  }
+
+  private static List<Class<?>> extractTypeParameters(Type fieldType) {
+    Type[] actualTypeArguments = ((ParameterizedType) fieldType).getActualTypeArguments();
+    List<Class<?>> typeParameters = new ArrayList<>();
+    Arrays.stream(actualTypeArguments)
+        .forEach(
+            (Type type) -> {
+              if (type instanceof ParameterizedType) {
+                typeParameters.addAll(extractTypeParameters(type));
+              } else {
+                typeParameters.add((Class<?>) type);
+              }
```

Comment:
You could use a `.map` and some streaming operator to end up with a collect `.toList` not to have an ArrayList from outside the stream flow. Also, an array list is mutable, take care of external mutations.